account_bank_statement_import/account_bank_statement_import.py

```diff
@@ -69,15 +72,35 @@ def import_file(self):
 
     @api.model
     def _import_file(self, data_file):
-        """ Create bank statement(s) from file
-        """
+        """ Create bank statement(s) from file."""
         # The appropriate implementation module returns the required data
-        currency_code, account_number, stmts_vals = self._parse_file(data_file)
-        # Check raw data
-        self._check_parsed_data(stmts_vals)
+        statement_ids = []
+        notifications = []
+        statements = self._parse_file(data_file)
```

Comment:
@lmignon I do not quite see how your refactoring would work. It is still having _parse_file with the same return value that I proposed. Or do I miss something?
@lmignon @hbrunn I am not quite agreed that the "existing API" argument is valid, as the whole banking import framework is quite new and everybody knows it was/is still in flux. Look at the whole refactoring for the 8.0 API. Then, adapting to the (IMHO better) return value in many cases will only need two or three lines of codes. Look at the changes I made to the ofx and qif import. They are absolutely minimal.